### PR TITLE
Various changes (mainly library management)

### PR DESCRIPTION
**Pull request by nad**
This pull request includes changes related to issues #6333 and #6465, plus some changes related to the parser.

### CODE REVIEW

File: src/full/Agda/Compiler/Common.hs [hunk 1]
```diff
@@ -168,15 +169,18 @@ inCompilerEnv checkResult cont = do
     -- Unfortunately, a pragma option is stored in the interface file as
     -- just a list of strings, thus, the solution is a bit of hack:
     -- We match on whether @["--no-main"]@ is one of the stored options.
-    when (["--no-main"] `elem` iFilePragmaOptions mainI) $
+    when (["--no-main"] `elem`
+          map pragmaStrings (iFilePragmaOptions mainI)) $
```
Comment (andreasabel) [inline src/full/Agda/Compiler/Common.hs]: Could you factor out `map pragmaStrings (iFilePragmaOptions mainI)`?  It appears several times subsequently and giving it a name would make the code more succinct and more readable.

File: doc/user-manual/tools/package-system.rst [hunk 2]
```diff
@@ -119,6 +119,12 @@ Note that if the search finds two or more ``.agda-lib`` files, then
 the flags from all of these files are used, and flags from different
 files are ordered in an unspecified way.
 
+Note also that there must not be any ``.agda-lib`` files below the
+root, on the path to the Agda file. For instance, if the top-level
+module in the Agda file is called ``A.B.C``, and it is in the
+directory ``root/A/B``, then there must not be any ``.agda-lib`` files
+in ``root/A`` or ``root/A/B``.
```
Comment (andreasabel) [inline doc/user-manual/tools/package-system.rst:126]: Makes sense.

**Comment by andreasabel** [comment]
This essentially 4 PRs, one for each commit.  Can't sensibly review this as "PR".

**Comment by nad** [comment]
> This essentially 4 PRs, one for each commit. Can't sensibly review this as "PR".

I did not ask for a review, I just wanted to run the CI tests, and if I recall correctly we can no longer do this without creating a PR.

File: src/full/Agda/Compiler/Common.hs (hunk 1 above)
Comment (nad) [reply src/full/Agda/Compiler/Common.hs]: OK.

**Comment by andreasabel** [comment]
> I did not ask for a review, I just wanted to run the CI tests, and if I recall correctly we can no longer do this without creating a PR.

You can get CI by prefixing your branch name with `ci-`, e.g `ci-issue-6465`.

However, we decided that changes to Agda should go through a PR always, to give the other developers a chance to review and a forum to discuss the changes.

**Pull request merged**